packages/next/build/utils.ts

```diff
@@ -1120,6 +1120,17 @@ export function getRawPageExtensions(pageExtensions: string[]): string[] {
   )
 }
 
+export function isPage(
+  filename: string,
+  pagesDir: string,
+  pageExtensions: string[]
+): boolean {
+  return (
+    filename.startsWith(pagesDir) &&
+    new RegExp(`\\.(?:${pageExtensions.join('|')})$`).test(filename)
```

Comment:
```suggestion
    filename.startsWith(pagesDir) &&
    (!pageExtensions || pageExtensions.some((ext) => filename.endsWith(ext)))
```
    
Could we do it without a RegExp? Slightly more readable.